[test_profiler.py] Disable tqdm monitor thread and torch.compile with compile_threads=1

shengfukevin opened the pull request:
Summary: if tqdm is not shutdown properly, it will leave the monitor thread alive. This causes an issue in the multithreading test because we check all events in that test with their tids. The events that correspond to these lingering threads all have TID of (uint64_t)(-1) which is invalid. The work around is turning off monitoring thread when tqdm is loaded. Since these are unit tests, it is safe to turn off monitor thread.

Test Plan: buck test  mode/dev-sand caffe2/test:profiler

Differential Revision: D56310301

aaronenyeshi commented on test/profiler/test_profiler.py:
```diff
@@ -1,4 +1,14 @@
+# threads all have TID of (uint64_t)(-1) which is invalid.
+# The work around is turnning off monitoring thread when tqdm is loaded.
+# Since these are unit tests, it is safe to turn off monitor thread.
+import tqdm
+tqdm.tqdm.monitor_interval = 0
```
lintrunner error
```suggestion
import tqdm

tqdm.tqdm.monitor_interval = 0
```

aaronenyeshi reviewed (changes requested):
please fix test failures:
```
2024-04-18T17:54:01.3338407Z Traceback (most recent call last):
2024-04-18T17:54:01.3339508Z   File "/var/lib/jenkins/workspace/test/profiler/test_profiler.py", line 9, in <module>
2024-04-18T17:54:01.3340577Z     import tqdm
2024-04-18T17:54:01.3341206Z ModuleNotFoundError: No module named 'tqdm'
```
Also, a lintrunner error.

aaronenyeshi reviewed (approved):
LGTM!

aaronenyeshi commented:
@pytorchbot rebase

aaronenyeshi commented:
@pytorchbot rebase

aaronenyeshi commented:
@pytorchbot merge

shengfukevin committed:
[test_profiler.py] Disable tqdm monitor thread and torch.compile with compile_threads=1 (#124409)

Summary:
if tqdm is not shutdown properly, it will leave the monitor thread alive. This causes an issue in the multithreading test because we check all events in that test with their tids. The events that correspond to these lingering threads all have TID of (uint64_t)(-1) which is invalid. The work around is turning off monitoring thread when tqdm is loaded. Since these are unit tests, it is safe to turn off monitor thread.


Test Plan: buck test  mode/dev-sand caffe2/test:profiler

Differential Revision: D56310301

Pulled By: aaronenyeshi
